### DISCUSSION TITLE
Adding all users to the dialout group

### DISCUSSION THREAD

**Discussion by danboid** [ltsp/ltsp#796] [Q&A] [answered]
I would like to add all LTSP users to the dialout out group so that users can program Arduinos via Arduino IDE. We're running Ubuntu 20.04 and LTSP5.

I have tried adding the following to ltsp.conf:

```
POST_INIT_DIALOUT="printf '# LTSP: allow access to Arduino, Wedo etc:\n*;*;*;Al0000-2400;dialout\n' >> /etc/security/group.conf && printf '# LTSP: allow access to Arduino, Wedo etc:\nauth\toptional\tpam_group.so\n' >> /etc/pam.d/common-auth"
```

and I've also tried it like this:

```
POST_INIT_DIALOUT="section_dialout"
[dialout]
printf '# LTSP: allow access to Arduino, Wedo etc:\n*;*;*;Al0000-2400;dialout\n' >> /etc/security/group.conf && printf '# LTSP: allow access to Arduino, Wedo etc:\nauth\toptional\tpam_group.so\n' >> /etc/pam.d/common-auth
```

but neither has worked. How are we supposed to add all users to a secondary group?

Thanks

**Comment by alkisg** [comment]
In the default LTSP setup, that's supposed to work.
You may also `cat /etc/pam.d/common-auth` after boot to make sure the line ended up there.
While if you're using a customized ldap setup, things could be different.

Another option that's probably better, but not well documented, is to use ACLs:
https://enotty.pipebreaker.pl/2012/05/23/linux-automatic-user-acl-management/

**Reply by danboid** [reply]
I suspect I'm going to have to try adding all users into the dialout group on the LDAP server because I've had no luck with groups.conf or using udev.

**Comment by Soylor** [comment]
Try with POST_SERVICE_DIALOUT instead of POST_INIT_DIALOUT

Or ypou can give permissions directily with a POST_INIT_x rule (or modify directily in image)

`sudoedit /etc/udev/rules.d/50-myusb.rules
`
Then save this text:
```
KERNEL=="ttyUSB[0-9]*",MODE="0666"
KERNEL=="ttyACM[0-9]*",MODE="0666"

```

**Reply by danboid** [reply]
Thanks Soylor! I shall try this out next week.

**Reply by danboid** [reply]
None of these solutions have worked so it looks like we'll have to try and add all the users to the dialout group on our domain controller.

**Reply by Soylor** [reply]
I got it working with POST_SERVICE_x
Tomorrow i’ll cut and paste solution

**Reply by danboid** [reply]
That's great news! Yes, please do. Thanks!

**Comment by Soylor** [comment, marked as answer]
`ltsp.conf
`
```
[clientes]
POST_SERVICE_DIALOUT="section_dialout"

[dialout]
echo "*; *;*; Al0000-2400;dialout" >> /etc/security/group.conf
echo "auth     optional        pam_group.so" >> /etc/pam.d/common-auth

```
Of course `sudo ltsp initrd `is needed.
Tha's all.

**Reply by danboid** [reply]
Thanks Soylor! That has fixed it!

I also had to remove the brltty package because its kernel modules clash with some Arduino's and stop them working.
